adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -539,6 +654,45 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   }
 
   /**
+   * All shuffleRegionJoins share the same code to prepare, so this is a helper
+   * function that prepares all the data for joining. It starts by repartitioning
+   * (if needed) the leftRdd. Next, it copartitions the right Rdd with it. Before
+   * returning the RDDs, it puts them in the format that the joins need. The
+   * ReferenceRegion is inferred for each individual record in the RDD.
+   *
+   * @param leftRdd Usually the object that invoked the function, unless a
+   *                repartition had to occur. The repartition only happens in cases where
```

Comment:
Stylistic nit: we've typically been preferring to indent lines in `@param`,`@return`, etc by 2 spaces.